ui/src/app/views/workspace-edit.tsx

```diff
@@ -641,23 +715,51 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
                     value={this.state.workspace.researchPurpose[rp.shortName]}
                     onChange={v => this.updateResearchPurpose(rp.shortName, v)}
                     children={rp.shortName === 'diseaseFocusedResearch' ?
+                    <div style={{position: 'relative'}}>
                       <TooltipTrigger
                         content='You must select disease focused research to enter
-                          a disease of focus'
-                        disabled={this.state.workspace.researchPurpose.diseaseFocusedResearch}>
+                            a disease of focus'
+                          disabled={this.state.workspace.researchPurpose.diseaseFocusedResearch}>
                         <TextInput value={this.state.workspace.researchPurpose.diseaseOfFocus}
-                          style={{
-                            width: 'calc(50% - 2rem)',
-                            border: '1px solid #9a9a9',
-                            borderRadius: '5px'
+                          style={{width: '90%', border: '1px solid #9a9a9', borderRadius: '5px'}}
+                          onChange={e => {
+                            this.searchTermChangedEvent(e);
+                            this.setState(fp.set([
+                              'workspace',
+                              'researchPurpose',
+                              'diseaseOfFocus'
+                            ], e));
                           }}
-                          placeholder='Name of Disease' onChange={v =>
-                          this.setState(
-                            fp.set(['workspace', 'researchPurpose', 'diseaseOfFocus'], v))}
+                          placeholder='Name of Disease'
                           disabled={!this.state.workspace.researchPurpose.diseaseFocusedResearch}/>
-                      </TooltipTrigger> : undefined}/>
-              )}
-            </div>
+                      </TooltipTrigger>
+                      {this.state.workspace.researchPurpose.diseaseFocusedResearch &&
+                       this.showSearchResults &&
+                        <div data-test-id='drop-down' style={{...styles.dropdownMenu,
+                          ...styles.open, minWidth: '90%'}}>
+                          {this.state.autocompleteDiseases.map((disease, j) => {
+                            return (
+                            <div key={j} style={this.state.diseaseHover[j] ?
+                              styles.boxHover : styles.box}
+                                onMouseOver={() =>
+                                  this.setState({diseaseHover: this.toggleDiseaseHover(j)})}
+                                onMouseOut={() =>
+                                  this.setState({diseaseHover: this.toggleDiseaseHover(j)})}
+                                onClick={() => {
+                                  this.setState(fp.set([
```

Comment:
Parsing this line, it looks like this will evaluate to a call to setState like:

this.setState({ workspace: { researchPurpose: { diseaseOfFocus: 'my-disease' } } });

I'm interested in the semantics here of how React handles setting state for a deeply-nested object. The docs say "shallow merge" (https://reactjs.org/docs/react-component.html#setstate), which suggests to me that this will cause the entire workspace object to be overwritten. That doesn't seem to be what we want to happen, but it also seems like it doesn't break the overall flow of this component.

If we wanted to be on the safe side, you could use the setState updater syntax to modify the latest version of the state object:

this.setState((prevState, props) => {
  fp.set('workspace.researchPurpose.diseaseOfFocus', disease, prevState);
  return {
    workspace: prevState.workspace,
    autocompleteDiseases: [],
  };
});

Reply:
The following experiment seems to indicate that the state is preserved.
```
    				onMouseDown={() => {
                                  this.setState((oldState, props) => {
                                    console.log("oldState:", oldState);
                                    const newState = fp.set([
                                      'workspace',
                                      'researchPurpose',
                                      'diseaseOfFocus'
                                    ], disease)(oldState);
                                    console.log("newState:", newState);
                                    return newState;
                                  });
```
Results:
```
oldState: 
{cdrVersionItems: Array(1), workspace: {…}, workspaceCreationConflictError: false, workspaceCreationError: false, workspaceCreationErrorMessage: "", …}
cdrVersionItems: [{…}]
cloneUserRole: false
diseaseEntryState: 8
diseaseHover: (20) [false, false, true, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
diseaseList: (20) ["lepromatous leprosy", "larynx leiomyosarcoma", "large bowel leiomyoma", "larynx leiomyoma", "leptospirosis", "laryngostenosis", "legionellosis", "lactocele", "labyrinthitis", "lagophthalmos", "leech infestation", "leptomeninges sarcoma", "Lafora disease", "leiomyomatosis", "Legionnaires' disease", "Leber congenital amaurosis", "cutaneous leishmaniasis", "LEOPARD syndrome", "Lambert-Eaton myasthenic syndrome", "leishmaniasis"]
loading: false
showStigmatizationDetails: false
showUnderservedPopulationDetails: false
workspace: {name: "", dataAccessLevel: "registered", namespace: undefined, cdrVersionId: "1", researchPurpose: {…}}
workspaceCreationConflictError: false
workspaceCreationError: false
workspaceCreationErrorMessage: ""
__proto__: Object
workspace-edit.tsx:791 newState: 
{cdrVersionItems: Array(1), workspace: {…}, workspaceCreationConflictError: false, workspaceCreationError: false, workspaceCreationErrorMessage: "", …}
cdrVersionItems: [{…}]
cloneUserRole: false
diseaseEntryState: 8
diseaseHover: (20) [false, false, true, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
diseaseList: (20) ["lepromatous leprosy", "larynx leiomyosarcoma", "large bowel leiomyoma", "larynx leiomyoma", "leptospirosis", "laryngostenosis", "legionellosis", "lactocele", "labyrinthitis", "lagophthalmos", "leech infestation", "leptomeninges sarcoma", "Lafora disease", "leiomyomatosis", "Legionnaires' disease", "Leber congenital amaurosis", "cutaneous leishmaniasis", "LEOPARD syndrome", "Lambert-Eaton myasthenic syndrome", "leishmaniasis"]
loading: false
showStigmatizationDetails: false
showUnderservedPopulationDetails: false
workspace: {name: "", dataAccessLevel: "registered", namespace: undefined, cdrVersionId: "1", researchPurpose: {…}}
workspaceCreationConflictError: false
workspaceCreationError: false
workspaceCreationErrorMessage: ""
__proto__: Object
```